packages/ui/src/Button.tsx

```diff
@@ -92,71 +100,101 @@ export function Button({
-      }}
-      onPress={debounce(
-        async () => {
-          Unifier.utils.haptic();
-          setLoading(true);
-          try {
-            if (onClick) {
-              await onClick();
+    <>
+      <TouchableOpacity
+        disabled={disabled || loading}
+        style={{
+          alignSelf: inline === true ? undefined : "stretch",
+          height: HEIGHTS[size || "md"],
+          backgroundColor: getBackgroundColor(color),
+          // width: inline === true ? undefined : "100%",
+          flexShrink: inline ? 1 : 0,
+          // flexGrow: inline ? 0 : 1,
+          alignItems: "center",
+          justifyContent: "center",
+          borderRadius: 5,
+          borderColor: getBorderColor(color),
+          borderWidth: type === "outline" ? 2 : 0,
+          opacity: disabled ? 0.4 : 1,
+          flexDirection: "row",
+          paddingHorizontal: 4 * 2,
+        }}
+        onPress={debounce(
+          async () => {
+            Unifier.utils.haptic();
+            setLoading(true);
+            try {
+              if (withConfirmation) {
+                if (!showConfirmation) {
+                  setShowConfirmation(true);
+                }
+              } else if (onClick) {
+                await onClick();
+              }
+            } catch (e) {
+              setLoading(false);
+              throw e;
             }
-          } catch (e) {
             setLoading(false);
-            throw e;
-          }
-          setLoading(false);
-        },
-        500,
-        {leading: true}
-      )}
-    >
-      {icon !== undefined && (
-        <Box paddingX={2}>
-          <Icon
+          },
+          500,
+          {leading: true}
+        )}
+      >
+        {icon !== undefined && (
+          <Box paddingX={2}>
+            <Icon
+              color={getTextColor(color as Color)}
+              name={icon}
+              prefix={iconPrefix || "far"}
+              size={size}
+            />
+          </Box>
+        )}
+        {Boolean(children) && children}
+        {Boolean(text) && (
+          <Text
             color={getTextColor(color as Color)}
-            name={icon}
-            prefix={iconPrefix || "far"}
+            font="button"
+            inline={inline}
             size={size}
-          />
-        </Box>
-      )}
-      {Boolean(children) && children}
-      {Boolean(text) && (
-        <Text
-          color={getTextColor(color as Color)}
-          font="button"
-          inline={inline}
-          size={size}
-          skipLinking
-          weight="bold"
-        >
-          {text}
-        </Text>
-      )}
-      {(loading || loading) && (
-        <Box marginLeft={2}>
-          <ActivityIndicator color={getTextColor(color as Color)} size="small" />
-        </Box>
-      )}
-    </TouchableOpacity>
+            skipLinking
+            weight="bold"
+          >
+            {text}
+          </Text>
+        )}
+        {(loading || loading) && (
```

Comment:
I know this isn't your code, but this could be just `loading` or `Boolean(loading)` for extra safety.